Cython/Compiler/FusedNode.py

```diff
@@ -303,9 +304,19 @@ def _fused_instance_checks(self, normal_types, pyx_code, env):
                 """)
 
     def _dtype_name(self, dtype):
+        # avoid duplicating names
+        if dtype in self.used_names:
+            return self.used_names[dtype]
         if dtype.is_typedef:
-            return '___pyx_%s' % dtype
-        return str(dtype).replace(' ', '_')
+            base_name = ('___pyx_%s' % dtype).replace(' ', '_')
+        else:
+            base_name = str(dtype).replace(' ', '_')
+        name = base_name
+        n = 0
+        while name in self.used_names:
+            name = base_name + str(n)
```

Comment:
This seems untested. It should say `n += 1` somewhere.

Reply:
Yes this wasn't working. I've got something that at least tests a single name clash - I can't immediately find a way to get it to go round the loop a second time though.